contribs/gmf/src/directives/datepicker.js

```diff
@@ -0,0 +1,170 @@
+
+
+/**
+ * Provide a directive to select a signle date or a range of dates
+ * @param {string|function(!angular.JQLite=, !angular.Attributes=)}
+ * gmfDatePickerTemplateUrl Template for the directive.
+ * @param  {angular.$timeout} $timeout angular timeout service
+ * @return {angular.Directive} The directive specs.
+ * @ngInject
+ * @ngdoc directive
+ * @ngname gmfDatePicker
+ */
+gmf.DatePicker = function(gmfDatePickerTemplateUrl,  $timeout) {
+  return {
+    scope : {
+      onDateSelected : '&',
+      time : '='
+    },
+    bindToController : true,
+    controller : 'gmfDatePickerController',
+    controllerAs : 'timeCtrl',
+    restrict: 'AE',
+    templateUrl : gmfDatePickerTemplateUrl,
+    link: function(scope, element, attrs, ctrl) {
+
+      /** @type {string} */
+      var dFormat;
+
+      var lang =  ctrl.gettextCatalog_.getCurrentLanguage();
+      $['datepicker']['setDefaults']($['datepicker']['regional'][lang]);
+
+      switch (lang) {
+        case 'de':
+          dFormat = 'dd.mm.yy';
+          break;
+        default:
+          dFormat = 'dd/mm/yy';
+          break;
+      }
+
+      var commondateOptions_ = {
+        'dateFormat' : dFormat,
+        'minDate' : new Date(ctrl.time.minValue),
+        'maxDate' : new Date(ctrl.time.maxValue)
+      };
+
+      ctrl.sdateOptions = angular.extend({}, commondateOptions_, {
+        'onClose' : function(selectedDate) {
+          if (selectedDate) {
+            $(element[0]).find('input[name="edate"]').datepicker('option', 'minDate', selectedDate);
+          }
+        }
+      });
+
+      ctrl.edateOptions = angular.extend({}, commondateOptions_, {
+        'onClose' : function(selectedDate) {
+          if (selectedDate) {
+            $(element[0]).find('input[name="sdate"]').datepicker('option', 'maxDate', selectedDate);
+          }
+        }
+      });
+
+      angular.element('body').on('hidden.bs.popover', function() {
+        var dp = angular.element('#ui-datepicker-div');
+        if (dp && dp.css('display') === 'block') {
+          $(element[0]).find('input[name$="date"]').datepicker('hide');
+        }
+      });
+
+      $timeout(function() {
+        angular.element('#ui-datepicker-div').on('mousedown', function(e) {
+          e.stopPropagation();
+        });
+      });
+    }
+  };
+};
+
+gmf.module.directive('gmfDatePicker', gmf.DatePicker);
+
+
+/**
+ * DatePickerController - directive conttroller
+ * @param {!angular.Scope} $scope Angular scope.
+ * @param {angular.$injector} $injector injector.
+ * @constructor
+ * @export
+ * @ngInject
+ * @ngdoc controller
+ * @ngname gmfDatePickerController
+ */
+gmf.DatePickerController = function($scope, $injector) {
+
+  /**
+   * The gettext catalog
+   * @type {angularGettext.Catalog}
+   * @private
+   */
+  this.gettextCatalog_ = $injector.get('gettextCatalog');
+
+
+  /**
+   * @function
+   * @export
+   */
+  this.onDateSelected;
+
+
+  /**
+   * @type {Object}
+   * @export
+   */
+  this.time;
```

Comment:
I called it 'time' because under the wood, it's the 'time' parameter from a node.